frontend/app/src/components/history/ledger-actions/LedgerActionForm.vue

```diff
@@ -114,24 +118,46 @@ const reset = () => {
 };
 
 const setEditMode = () => {
-  const ledgerAction = get(edit);
+  const ledgerAction = get(formData);
   if (!ledgerAction) {
     reset();
     return;
   }
 
   set(location, ledgerAction.location);
-  set(datetime, convertFromTimestamp(ledgerAction.timestamp, true));
+  if (ledgerAction.timestamp) {
+    set(datetime, convertFromTimestamp(ledgerAction.timestamp, true));
+  } else {
+    set(datetime, convertFromTimestamp(dayjs().unix(), true));
+  }
   set(asset, ledgerAction.asset);
-  set(amount, ledgerAction.amount.toFixed());
-  set(actionType, ledgerAction.actionType.toString());
+  if (ledgerAction.amount) {
+    set(amount, ledgerAction.amount.toFixed());
+  } else {
+    set(amount, '0');
+  }
+  if (ledgerAction.actionType) {
+    set(actionType, ledgerAction.actionType.toString());
+  } else {
+    set(actionType, LedgerActionType.ACTION_INCOME);
+  }
   set(rate, ledgerAction.rate?.toFixed() ?? '');
   set(rateAsset, ledgerAction.rateAsset ?? '');
   set(link, ledgerAction.link ?? '');
   set(notes, ledgerAction.notes ?? '');
   set(id, ledgerAction.identifier);
 };
 
+const ledgerActionStore = useLedgerActionStore();
+const { addLedgerAction, editLedgerAction } = ledgerActionStore;
+
+const saveData = async (ledgerAction: NewLedgerAction | LedgerActionEntry) => {
+  if ((ledgerAction as LedgerActionEntry).identifier) {
```

Comment:
if you do `indentifier` in ledgerAction should help narrow the type and it will allow us to remove the casting. better do proper narrowing as casting can sometimes hide bugs